auth/src/main/java/com/firebase/ui/auth/ui/phone/SubmitConfirmationCodeFragment.java

```diff
@@ -139,41 +166,19 @@ public void onDestroy() {
         mLooper.removeCallbacks(mCountdown);
     }
 
-    private void setupSubmitConfirmationButton() {
-        mSubmitConfirmationButton.setEnabled(false);
-        mSubmitConfirmationButton.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                submitCode();
-            }
-        });
-    }
-
     private void setupConfirmationCodeEditText() {
         mConfirmationCodeEditText.setText("------");
         mConfirmationCodeEditText.addTextChangedListener(new BucketedTextChangeListener(
-                mConfirmationCodeEditText, 6, "-",
+                mConfirmationCodeEditText, VERIFICATION_CODE_LENGTH, "-",
                 new BucketedTextChangeListener.ContentChangeCallback() {
                     @Override
-                    public void whileComplete() {
-                        mSubmitConfirmationButton.setEnabled(true);
+                    public void whenComplete() {
```

Comment:
My only concern here is that this is less typo-friendly.  If you accidentally type the wrong 6 digits (maybe you fat-finger a duplicate digit in the middle) then it's going to submit automatically.

For instance I just entered the code and got into this state:
![image](https://user-images.githubusercontent.com/8466666/50651999-ed484600-0f39-11e9-84c5-ffb6af471c7f.png)

There are two UX issues here:
  * The error message position is strange
  * It's not obvious that entering that last digit was "submitting" and I should back out and proceed.

I think the second issue could be mitigated by deleting the digits entered when an error happens?  Although that could be jarring.

Reply:
Yeah, you're right, we're optimizing for the successful case (which should be more common).

For the error message, I think that's out of scope for this PR since we'll have to use some [ugly hacks](https://stackoverflow.com/questions/42575825/textinputlayout-error-right-align). As for typos, I like the idea of nuking the text. 👍